### ISSUE TITLE
Badge request: OpenHub

### ISSUE THREAD

**Issue by Paebbels** [badges/shields#4480]
:clipboard: **Description**

Please provide badges for OpenHub, because the provided badges from OpenHub do not fit among other badges (from Shields.io) in README files.

Open Hub is a statistics website, that scans registered open source projects e.g. on GitHub and creates global statistics per user and project. E.g. it aggregates all commits a person has ever done in any repository. It indexes 495k repositories and counts 4.9 million contributors.

Link: https://www.openhub.net/

**Example badges from Open Hub:**

1. ![](https://www.openhub.net/accounts/Paebbels/widgets/account_detailed?format=gif&amp;ref=sample)
2. ![](https://www.openhub.net/accounts/Paebbels/widgets/account_rank?format=gif&amp;ref=sample)
3. ![](https://www.openhub.net/accounts/Paebbels/widgets/account_tiny?format=gif&amp;ref=sample)

**What could be done by Shields.io?**

* `Open Hub | rank 7 `  
  ![](https://img.shields.io/badge/OpenHub-rank%207-blue)
* `Open Hub | 1059 commits`  
  ![](https://img.shields.io/badge/OpenHub-1059%20commits-blue)

**Doing more:**
* per user
  * active since / first commit
  * total commits
  * most recent commit
  * contributed to X projects
* per project
  * years of effort (based on COCOMO)
  * estimated project value in USD (based on COCOMO)
  * most used language per project

:link: **Data**

* API documentation at https://github.com/blackducksoftware/ohloh_api#ohloh-api-documentation

**Further resources:**
* https://github.com/blackducksoftware/ohloh-ui

:microphone: **Motivation**

* The provided badges from OpenHub do not fit among other badges in README files.
* FOSS contributors might want to publish their aggregated statistics gathered from multiple repositories and repository platforms on personal websites or profile sites.

**Comment by calebcartwright** [comment]
Thanks for the request @Paebbels! This seems reasonable, though it would require an account creation and Shields being granted an increased api limit since they are throttling api consumers at 1,000 requests per day

> Before you can access the Ohloh API, you must register your application and obtain an API key. Bandwidth will initially be limited to 1,000 requests per API key per day

**Comment by paulmelnikow** [comment]
If someone is able to reach out to OpenHub and get more quota for Shields, let's consider this at that time.

**Comment by Paebbels** [comment]
I created this issue at OpenHub: https://github.com/blackducksoftware/ohloh-ui/issues/1491

**Comment by paulmelnikow** [comment]
I'm happy to say that Shields has an API key for Open Hub with a 10k request limit. Thank @PDegenPortnoy and @Paebbels for your help with that!

If anyone is interested in working on the implementation, a good place to start is our tutorial: https://github.com/badges/shields/blob/master/doc/TUTORIAL.md

Feel free to post questions here!

**Comment by PDegenPortnoy** [comment]
Please note that the Open Hub API key should be kept private to ensure that Shields.io and only Shields.io can use it. In Open Hub Land we will be starting work in a few days to design and implement badges that conform to the existing standards so there is a more direct means of getting ones information in the standard form.

**Comment by paulmelnikow** [comment]
Yup! This one will only be on the production servers. Only the folks running those servers will have access.

**Comment by Paebbels** [comment]
@paulmelnikow any progress on this issue?

**Comment by JoeIzzard** [comment]
I've started working on this now.

**Comment by Paebbels** [comment]
That's great !

------------
/cc @eine

**Comment by evilaliv3** [comment]
Super cool @JoeIzzard ! At @GlobaLeaks we would love to use that! thank you

**Comment by joshuacwnewton** [comment]
I just popped in to say that a project I'm working on is using this badge: 

```
https://www.openhub.net/p/<project-name>/widgets/project_thin_badge.gif
```

For example: ![Firefox](https://www.openhub.net/p/firefox/widgets/project_thin_badge.gif)

Which fits in with Shields.io badges much better. Was this considered? The 3 badges in the issue description are a bit different from this one.
